test/new-e2e/tests/agent-subcommands/diagnose/diagnose_common_test.go

```diff
@@ -20,6 +21,52 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// Diagnosis contains the results of the diagnosis
+type Diagnosis struct {
+	// --------------------------
+	// required fields
+
+	// run-time (pass, fail etc)
+	Result int `json:"result"`
+	// static-time (meta typically)
+	Name string `json:"name"`
+	// run-time (actual diagnosis consumable by a user)
+	Diagnosis string `json:"diagnosis"`
+
+	// --------------------------
+	// optional fields
+
+	// static-time (meta typically)
+	Category string `json:"category,omitempty"`
+	// static-time (meta typically, description of what being tested)
+	Description string `json:"description,omitempty"`
+	// run-time (what can be done of what docs need to be consulted to address the issue)
+	Remediation string `json:"remediation,omitempty"`
+	// run-time
+	RawError string `json:"rawerror,omitempty"`
+}
+
+// DiagnoseResult contains the results of the diagnose command
+type DiagnoseResult struct {
+	Diagnoses []Diagnoses `json:"diagnoses"`
+	Summary   Counters    `json:"summary"`
+}
+
+// Diagnoses is a collection of Diagnosis
+type Diagnoses struct {
+	SuiteName      string      `json:"suite_name"`
+	SuiteDiagnoses []Diagnosis `json:"diagnoses"`
+}
+
+// Counters contains the count of the diagnosis results
+type Counters struct {
+	Total         int `json:"total,omitempty"`
+	Success       int `json:"success,omitempty"`
+	Fail          int `json:"fail,omitempty"`
+	Warnings      int `json:"warnings,omitempty"`
+	UnexpectedErr int `json:"unexpected_error,omitempty"`
+}
```

Comment:
Any reason we do not import the struct from the diagnose package?

Reply:
Did not finish was just for test the idea is just to keep the field need to avoid importing the whole one